specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,137 @@
+|None | AnimatedVisualPlayer will stop the player and destroy animations when the player is idle. |
+
+If you have an animation that will start on click, you may want to set CacheMode to 'Always' when the mouse enters this control, 
+and back to 'None' when mouse leaves to ensure that the animation starts immediately on click. 
+
+Setting the CacheMode to Always when the mouse enters the control will ensure that the animation objects are
+preloaded so that the animation may start immediately when it needs to play. 
+
+```c#
+private void Player_PointerEntered(object sender, PointerRoutedEventArgs e)
+{
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode::Always;
+}
+async private void Player_PointerExited(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode::None;
+}
+
+async private void Player_OnClick(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.PlayAsync(0.0, 1.0);
+}
+
+```
+If 'PlayAsync' is called and the animations have not been instantiated due to CacheMode being set to 'None',
+'PlayAsync' will instantiate the animations so it can play the visual. 
+
+# API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.UI.Xaml.Controls
+{
+  unsealed runtimeclass AnimatedVisualPlayer
+  {
+      // Existing ...
+
+    //Sets the caching behavior of the AnimatedVisualPlayer
+    Microsoft.UI.Xaml.UIElement.CacheMode AnimationsCacheMode CacheMode;
+  }
+}
+```
+
+## IAnimatedVisual2 Interface
+
+An animated Composition.Visual that can be used by other objects such as an AnimtedVisualPlayer or AinmatedIcon. 
+
+| Method | Meaning |
+|-------------|---------|
+| InstantiateAnimations | Instantiates the animation visuals for the provided progress value of the animation. |
+| DestryAnimations | Destroys all animation visuals for a given animation.|
+
+```c# 
+interface IAnimatedVisual2 requires IAnimatedVisual
+{
+    void InstantiateAnimations(Double progressHint);
+    void DestroyAnimations();
+};
+
+```
+
+## IAnimatedVisual2.InstantiateAnimations Method
+
+Instantiates the animation visuals for the provided progress value of the animation. 
+
+```c# 
+void InstantiateAnimations(Double progressHint);
+```
+
+## IAnimatedVisual2.DestroyAnimations Method
+
+Destroys all animation visuals for a given animation. 
+
+```c# 
+void DestroyAnimations();
+```
+## IAnimatedVisualSource3 Interface
+
+An animated Visual that can be used by other objects, such as an AnimatedVisualPlayer.
+
+| Property | Meaning |
+|-------------|---------|
+| InstantiateAnimations | Instantiates the animation visuals for the animation. If set to false, no animations are instantiated.  |
+
+```c# 
+interface IAnimatedVisualSource3
+{
+    IAnimatedVisual2 TryCreateAnimatedVisual(
+        Windows.UI.Composition.Compositor compositor,
+        Boolean instantiateAnimations,
```

Comment:
This is essentially an overloaded method when combined with IAnimatedVisualSource, so the new parameter (`instantiateAnimations`) should be added at the end.

Reply:
Updated to put new method last.